crates/dap_adapters/src/go.rs

```diff
@@ -437,18 +439,44 @@ impl DebugAdapter for GoDebugAdapter {
             adapter_path.join("dlv").to_string_lossy().to_string()
         };
 
-        let cwd = task_definition
-            .config
-            .get("cwd")
-            .and_then(|s| s.as_str())
-            .map(PathBuf::from)
-            .unwrap_or_else(|| delegate.worktree_root_path().to_path_buf());
+        let cwd = Some(
+            task_definition
+                .config
+                .get("cwd")
+                .and_then(|s| s.as_str())
+                .map(PathBuf::from)
+                .unwrap_or_else(|| delegate.worktree_root_path().to_path_buf()),
+        );
 
         let arguments;
         let command;
         let connection;
 
         let mut configuration = task_definition.config.clone();
+
+        let mut envs = HashMap::default();
+        if let Some(env_file) = configuration.get("envFile").and_then(Value::as_str) {
+            if let Some(path) = {
+                if env_file.starts_with('/') {
+                    PathBuf::from_str(env_file).ok()
+                } else {
+                    cwd.as_ref().map(|p| p.clone().join(env_file))
+                }
+            } && let Ok(file) = delegate.fs().open_sync(&path).await
+            {
+                envs = dotenvy::from_read_iter(file)
+                    .filter_map(Result::ok)
+                    .collect();
+            } else {
+                warn!("Couldn't read env file: {env_file}");
```

Comment:
Can we make it a bit more obvious as to what's going on here? E.g. by stating that the env file could not be read while trying to start up a Go debugging session.